olp-cpp-sdk-core/include/olp/core/client/CancellationContext.h

```diff
@@ -29,58 +30,62 @@ namespace olp {
 namespace client {
 
 /**
- * @brief Wrapper which manages cancellation state for multiple network
+ * @brief Wrapper which manages cancellation state for any asynchronous
  * operations in a thread safe way.
+ * All public APIs are thread safe.
+ * This class is both movable and copyable.
  */
 class CORE_API CancellationContext {
  public:
-  /**
-  * @brief Execute a given cancellable code block if the operation has not yet
-  been cancelled. Otherwise, execute a custom cancellation function.
-  * @param execute_fn A function to execute if this operation is not yet
-  cancelled. This function should return a CancellationToken which
-  CancellationContext will propogate a cancel request to.
-  * @param cancel_fn A function which will be called if this operation is
-  already cancelled.
-  */
-  inline void ExecuteOrCancelled(
```

Comment:
Maybe ExecuteIfNotCancelled ?

Reply:
As discussed, I just marked the method as deprecated and it will be removed after TaskScheduler is in use.